ui/contexts/AppContext.tsx

```diff
@@ -1,45 +1,75 @@
+import _ from "lodash";
 import * as React from "react";
-import { Applications } from "../lib/api/applications/applications.pb";
+import { useHistory, useLocation } from "react-router-dom";
+import { Applications, User } from "../lib/api/applications/applications.pb";
+import { getToken } from "../lib/storage";
+import { PageRoute } from "../lib/types";
 
-type AppState = {
-  error: null | { fatal: boolean; message: string; detail?: string };
-};
+// Due to the way the grpc-gateway typescript client is generated,
+// we need to wrap each individual call to ensure the auth header gets
+// injected into the underlying `fetch` requests.
+// This saves us from having to rememeber to pass it as an arg in every request.
+function wrapClient<T>(client: any): T {
+  const wrapped = {};
+
+  _.each(client, (func, name) => {
+    wrapped[name] = (payload, options: RequestInit = {}) => {
+      const token = getToken();
```

Comment:
Check for no token here?

Reply:
Passing through empty string/null to the `Authorization` header is probably fine for now.